dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ContentletIndexAPIImpl.java

```diff
@@ -758,37 +758,43 @@ public BulkRequest appendBulkRequest(BulkRequest bulkRequest, final ReindexEntry
+        final List<ContentletVersionInfo> versions = APILocator.getVersionableAPI().findContentletVersionInfos(idx.getIdentToIndex());
         final Map<String, Contentlet> inodes = new HashMap<>();
-
-        for (ContentletVersionInfo cvi : versions) {
-            final String workingInode = cvi.getWorkingInode();
-            final String liveInode = cvi.getLiveInode();
-            inodes.put(workingInode, APILocator.getContentletAPI().findInDb(workingInode).orElse(null));
-            if (UtilMethods.isSet(liveInode) && !inodes.containsKey(liveInode)) {
-                inodes.put(liveInode, APILocator.getContentletAPI().findInDb(liveInode).orElse(null));
+        try {
+            for (final ContentletVersionInfo cvi : versions) {
+                final String workingInode = cvi.getWorkingInode();
+                final String liveInode = cvi.getLiveInode();
+                inodes.put(workingInode, APILocator.getContentletAPI().findInDb(workingInode).orElse(null));
+                if (UtilMethods.isSet(liveInode) && !inodes.containsKey(liveInode)) {
+                    inodes.put(liveInode, APILocator.getContentletAPI().findInDb(liveInode).orElse(null));
+                }
             }
-        }
-        inodes.values().removeIf(Objects::isNull);
-        if (inodes.isEmpty()) {
-            //If there is no content for this entry, it should be deleted to avoid future attempts that will fail also
-            APILocator.getReindexQueueAPI().deleteReindexEntry(idx);
-            Logger.debug(this, "unable to find versions for content id:" + idx.getIdentToIndex());
-        }
-        for (Contentlet contentlet : inodes.values()) {
-            Logger.debug(this, "indexing: id:" + contentlet.getInode() + " priority: " + idx.getPriority());
-            contentlet.setIndexPolicy(IndexPolicy.DEFER);
-
-            try {
+            inodes.values().removeIf(Objects::isNull);
+            if (inodes.isEmpty()) {
+                // If there is no content for this entry, it should be deleted to avoid future attempts that will fail also
+                APILocator.getReindexQueueAPI().deleteReindexEntry(idx);
+                Logger.debug(this, String.format("Unable to find versions for content id: '%s'. Deleting content " +
+                        "reindex entry.", idx.getIdentToIndex()));
+            }
+            for (final Contentlet contentlet : inodes.values()) {
+                Logger.debug(this, String.format("Indexing id: '%s', priority: '%s'", contentlet.getInode(), idx
+                        .getPriority()));
+                contentlet.setIndexPolicy(IndexPolicy.DEFER);
                 addBulkRequest(bulk, ImmutableList.of(contentlet), idx.isReindex());
-
-            } catch (Exception e) {
-                APILocator.getReindexQueueAPI().markAsFailed(idx, e.getMessage());
-
             }
+        } catch (final Exception e) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5458294)